evo_researcher/benchmark/utils.py

```diff
@@ -19,6 +20,39 @@ class PredictionResult(BaseModel):
     cost: t.Optional[float] = None
 
 
+AgentPredictions = t.Dict[str, PredictionResult]
+Predictions = t.Dict[str, AgentPredictions]
+
+
+class PredictionsCache(BaseModel):
+    predictions: Predictions
+
+    def save(self, path: str):
+        # If the file exists, load it and add the new predictions
+        if os.path.exists(path):
+            old_cache = self.parse_file(path)
```

Comment:
Sorry I stole your branch, as this is super helpful :-D But there is a bug, because it's loading old json here, if you have an existing cache file, and add a new agent, this will fail below on `old_cache.predictions[agent][question] = prediction` (as the agent from `self.predictions` isn't in the loaded cache dictionary).

It is probably an easy fix to add `old_cache.predictions[agent] =[]` here, but using this as a class instead of converting from/to dictionaries sounds less error-prone.